src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateClone = Object.assign({}, state);
+
+  actions.forEach(function (element) {
+    if (element.type === 'clear') {
+      stateClone = {};
+      stateHistory.push({ ...stateClone });
+    }
+
+    if (element.type === 'addProperties') {
+      stateClone = { ...stateClone, ...element.extraData };
+      stateHistory.push({ ...stateClone });
+    }
+
+    if (element.type === 'removeProperties') {
+      element.keysToRemove.forEach(function (key) {
+        delete stateClone[key];
+      });
+
+      stateHistory.push({ ...stateClone });
+    }
```

Comment:
Instead of using multiple if statements, consider using a switch statement as it is more suitable for a situation where you have a limited amount of conditions to check. Additionally, include a default case in your switch statement for error handling.